published/009-2016-March-28.md

```diff
@@ -0,0 +1,239 @@
+# IPFS Weekly #9
+
+[IPFS](//ipfs.io/) is a new hypermedia distribution protocol, addressed by content and identities, aiming to make the web faster, safer, and more open. In these posts, we highlight some of the development that has happened in the past week. For anyone looking to get involved, follow the embedded hyperlinks, search the wealth of information on [GitHub](//github.com/ipfs) or join us on [IRC](//webchat.freenode.net/?channels=ipfs) (#ipfs on the Freenode network).
+
+If you would like to get this update as an email, sign up for our [weekly newsletter](//tinyletter.com/ipfsweekly)!
+
+Here are some of the highlights for the [March 28th](//github.com/ipfs/pm/issues/98) sprint:
+
+## Updates
+
+### Package Managers
+
+Package managers have been a large topic of discussion recently. Mostly, this is because of an issue with an author of some heavily used [npm](https://npmjs.com) packages unpublishing all of his modules simultaneously. One of these was `left-pad`, which was used by thousands of builds globally, all of which broke when the package was removed. A great writeup of what happened is on the [npm.js blog here](http://blog.npmjs.org/post/141577284765/kik-left-pad-and-npm); they took this very seriously, and shortly after [changed their unpublish policy](http://blog.npmjs.org/post/141905368000/changes-to-npms-unpublish-policy) as a direct result.
+
+Many people jumped to IPFS as a possible solution to this problem. With a permanent filesystem, unpublishing wouldn't be possible. Here's one post titled [How to use IPFS to fix npm](http://amreldib.com/blog/HowToUseIpfsToFixNpm); here's [an issue](https://github.com/ipmjs/ipmjs/issues/9) on a new GitHub organization, ipmjs, trying to find consensus on how to fix npm using a permanent storage system; here's an npm module, [cowpen](https://www.npmjs.com/package/cowpen) that publishes modules directly to IPFS; here's [another decentralized package manager](https://github.com/mhhf/spore) that sprung up using IPFS and Ethereum.
+
+The IPFS community has been thinking about immutable package managers for a long time. @diasdavid has a project called `registry-mirror`, which allows you to run an npm registry locally that is backed by packages retrieved from IPFS instead of NPM directly. He's written about a presentation he gave for it, [here](http://blog.daviddias.me/2015/12/08/stellar-module-management); the source code is [here](https://github.com/diasdavid/registry-mirror).
+
+On a similar note, [gx](//github.com/whyrusleeping/gx), a package manager for Go, was also mentioned in a lot of the discussions about npm and package managers, especially on [Hacker News](https://news.ycombinator.com/). In the past two weeks, the project has recieved over 1000 stars - before, there were around 50.
```

Comment:
- maybe link directly to the gx HN post
- maybe say "the project went from 50 to 1000 stars, so people are interested in this now."